src/Microsoft.Framework.Runtime.Hosting/Json/JsonDeserializer.cs

```diff
@@ -0,0 +1,179 @@
+            {
+                return DeserializeObject(next, buffer);
+            }
+
+            if (next.Type == JsonTokenType.String)
+            {
+                return new JsonString(next.Value, next.Line, next.Column);
+            }
+
+            if (next.Type == JsonTokenType.True || next.Type == JsonTokenType.False)
+            {
+                return new JsonBoolean(next);
+            }
+
+            if (next.Type == JsonTokenType.Null)
+            {
+                return new JsonNull(next.Line, next.Column);
+            }
+
+            if (next.Type == JsonTokenType.Number)
+            {
+                return new JsonNumber(next);
+            }
+
+            throw new JsonDeserializerException(JsonDeserializerResource.Format_UnexpectedToken(next.Value, next.Type), next);
+        }
+
+        private static JsonArray DeserializeArray(JsonToken head, JsonBuffer buffer)
+        {
+            var list = new List<JsonValue>();
+            while (true)
+            {
+                var next = buffer.Read();
+                if (next.Type == JsonTokenType.RightSquareBracket)
+                {
+                    break;
+                }
+
+                list.Add(DeserializeInternal(next, buffer));
+
+                next = buffer.Read();
+                if (next.Type == JsonTokenType.EOL)
+                {
+                    throw new JsonDeserializerException(
+                        JsonDeserializerResource.Format_InvalidSyntaxExpectation("JSON array", ']', ','),
+                        next);
+                }
+                else if (next.Type == JsonTokenType.RightSquareBracket)
+                {
+                    break;
+                }
+                else if (next.Type != JsonTokenType.Comma)
+                {
+                    throw new JsonDeserializerException(
+                        JsonDeserializerResource.Format_InvalidSyntaxExpectation("JSON array", ','),
+                        next);
+                }
+            }
+
+            return new JsonArray(list.ToArray(), head.Line, head.Column);
+        }
+
+        private static JsonObject DeserializeObject(JsonToken head, JsonBuffer buffer)
+        {
+            var dictionary = new Dictionary<string, JsonValue>();
+
+            // Loop through each JSON entry in the input object
+            while (true)
+            {
+                var next = buffer.Read();
+                if (next.Type == JsonTokenType.EOL)
+                {
+                    throw new JsonDeserializerException(
+                        JsonDeserializerResource.Format_InvalidSyntaxExpectation("JSON object", '}'),
+                        next);
+                }
+
+                if (next.Type == JsonTokenType.Colon)
+                {
+                    throw new JsonDeserializerException(
+                        JsonDeserializerResource.Format_InvalidSyntaxNotExpected("JSON object", ':'),
```

Comment:
This is informational enough. JSON syntax is easy enough to figure out. And the exception message explain exactly what's the issue it runs into.